src/app/data/AdData.jsx

```diff
@@ -0,0 +1,229 @@
+import { containsEmail, extractEmail, formatDate, isValidEmail, mailtoInString } from "@/app/_common/utils/utils";
+import DOMPurify from "isomorphic-dompurify";
+import fixLocationName from "@/app/_common/utils/fixLocationName";
+import { parseISO } from "date-fns";
+
+/**
+ *  --------------------------- Ad Data ---------------------------
+ */
+export default function mapAdData(queryData) {
+    const data = queryData._source ? queryData._source : queryData ? queryData : undefined;
+    if (!data) {
+        return undefined;
+    }
+    const { properties } = data;
+    if (!properties) {
+        return undefined;
+    }
+
+    const adText = getAdText(properties.adtext);
+
+    return {
+        id: getString(data, "id"),
+        status: getString(data, "status"),
+        title: getString(data, "title"),
+        adText: getAdText(properties.adtext),
+        published: getString(data, "published"),
+        expires: getString(data, "expires"),
+        updated: getDate(data, "updated"),
+        source: getString(data, "source"),
+        reference: getString(data, "reference"),
+        medium: getString(data, "medium"),
+        applicationDue: getString(properties, "applicationdue"),
+        applicationEmail: getEmail(properties.applicationemail),
+        applicationUrl: getUrl(properties.applicationurl),
+        sourceUrl: getUrl(properties.sourceurl),
+        hasSuperraskSoknad: getString(properties, "hasInterestform"),
+
+        // employment details
+        engagementType: getString(properties, "engagementType"),
+        extent: getString(properties, "extent"),
+        jobArrangement: getString(properties, "jobarrangement"),
+        jobPercentage: getJobPercentage(properties.jobpercentage),
+        jobTitle: getString(properties, "jobtitle"),
+        positionCount: getString(properties, "positioncount"),
+        remote: getString(properties, "remote"),
+        sector: getString(properties, "sector"),
+        startTime: getDate(properties, "starttime"),
+        workday: getWorktime(properties.workday),
+        workhours: getWorktime(properties.workhours),
+        workLanguages: getArray(properties, "workLanguage"),
+        locationList: getLocationListData(data.locationList),
+        location: getString(properties, "location"),
+
+        // Employer
+        employer: getEmployerData(data),
+        contactList: getContactList(data.contactList),
+    };
```

Comment:
Bør vi legge de to dato propertiene vi bruker suppressHydrationWarning på i dag? Dvs publisert: ny i dag/i går/to dager siden samt frist: søk senest i dag/morgen... Da sier vi evt at det er datoen på serveren som gjelder, ikke på klienten.

Reply:
Jeg tenker at dato formatering kanskje ikke hører til i data layer?